content/solana-nft.md

```diff
@@ -0,0 +1,637 @@
+            "presale": true,
+            "discountPrice": 0.5
+        }
+    ```
+
+
+ - `hiddenSettings` can be used for hide-and-reveal drops (where the image of an NFT is revealed after the mint is complete). Using the `hiddenSettings` is outside the scope of this lesson, but you can read more about it [here](https://docs.metaplex.com/candy-machine-v2/configuration).
+
+    ```json
+        "hiddenSettings": {
+            "name": "My Hidden Collection ",
+            "uri": "uri",
+            "hash": "44kiGWWsSgdqPMvmqYgTS78Mx2BKCWzd"
+        }
+    ```
+
+ - `storage` indicates the storage type to upload images and metadata. Note that Arweave files are only stored for seven days on Devnet (When deploying to Mainnet use `arweave-sol`). You can the review list of supported storage types [here](https://docs.metaplex.com/candy-machine-v2/configuration).
+
+ - `noRetainAuthority` indicates whether the Candy Machine authority has the update authority for each mint or if it is transferred to the minter. This should be kept as `false` for the vast majority of cases.
+
+ - `noMutable` indicates whether the NFTs' metadata is mutable or not after having been minted. If set to false, the Candy Machine configurations can be updated. If set to true, Candy Machine configurations cannot be updated and cannot be reset to true.
+
+# Demo
+
+Let’s put all of this into practice by creating a Candy Machine and minting our collection using the Candy Machine UI.
+
+### 1. Download the starter code
+
+**Demo Starter Code:** [https://github.com/ZYJLiu/metaplex-starter](https://github.com/ZYJLiu/metaplex-starter)
+
+Let's begin by downloading the starter code. The starter code includes an "assets" folder and the configuration file for our Candy Machine, "config.json".
+
+### 2. Setup
+
+Next let's set up a new keypair to use for this lesson and connect to Devnet using the endpoint provided by Metaplex.
+
+From within the `metaplex-starter` directory, generate a new keypair to use for the project:
+
+```sh
+solana-keygen new --outfile ~/.config/solana/devnet.json
+```
+
+Set the new keypair as the default keypair:
+
+```sh
+solana config set --keypair ~/.config/solana/devnet.json
+```
+
+Set our connection to Devnet using Metaplex RPC:
+
+```sh
+solana config set --url https://metaplex.devnet.rpcpool.com/
+```
+
+Airdrop Devnet SOL to our new test wallet:
+
+```sh
+solana airdrop 2
+```
+
+Copy the keypair from `.config/solana/devnet.json` and import it into Phantom
+
+![Gif of Phantom Import](../assets/solana-nft-phantom.gif)
+
+### 3. Download Metaplex
+
+While in the `metaplex-starter` directory, clone the Metaplex repo:
+
+```sh
+git clone https://github.com/metaplex-foundation/metaplex.git
+```
+
+Next, install Metaplex dependencies:
+
+```sh
+yarn install --cwd metaplex/js/
+```
+
+Confirm that Candy Machine was installed:
+
+```sh
+ts-node metaplex/js/packages/cli/src/candy-machine-v2-cli.ts --version
```

Comment:
This doesn't work on my machine. I'm guessing it's either because of being on a Mac or there's a configuration issue. Either way, we should find a way to do this that has broader coverage

Reply:
K turns out I didn't have `ts-node` installed globally. It may be worth noting somewhere before this that if they don't already have this installed globally they should do:
```
npm install -g typescript
npm install -g ts-node
```

Now outside of that issue, I still can't get the cli to work. Debugging now but you can implement the rest of my comments and I'll figure this out.